BPB: Allow for multiple instances

obenland opened the pull request:
#### Changes proposed in this Pull Request

* Allows for multiple instances of the Blog Posts Block.

This was fixed on the Newspack side a while ago, but it looks like we missed removing this restriction on our side.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Checkout this PR and build the blocks with `$ npx lerna run dev --scope='@automattic/full-site-editing' --stream --no-prefix` from the `/apps/full-site-editing` folder.
* In a new or existing post or page, attempt to insert more than one Blog Posts block.

Fixes #39251

obenland commented:
@noahtallen Is that "Send hook to Mission Control" failure something I need to worry about?

jeffersonrabb commented:
> AFAIR, we've disabled multiple instances for Blog Posts Block because of the editor vs. front-end disparity issue, which doesn't seem to be fixed yet (paYJgx-jW-p2#comment-428). I'm able to add multiple blocks when testing, but post duplicates are not being removed in the editor as they are in the front-end.

The change was merged on January 15 (https://github.com/Automattic/newspack-blocks/pull/289), and should be in all releases dating back to 1.0.0-alpha.22 (January 20). It's been working correctly on all Newspack sites since then. To confirm, are you certain you're testing with a version of the `newspack-blocks` codebase newer than this? If so, maybe @georgeh could take a look and see if there's an incompatibility with the version in FSE?

WunderBart commented:
Thanks for the quick response, @jeffersonrabb!

> To confirm, are you certain you're testing with a version of the `newspack-blocks` codebase newer than this?

I'm testing on the Homepage Articles v1.0.0-alpha.26 (currently synchronized to BPB, as in this PR).

> maybe @georgeh could take a look and see if there's an incompatibility with the version in FSE?

I'd love that. Maybe we missed some steps in setting this block up after synchronization? I remember registering the QueryStore to make the synced code work - it's done in [editor.js](https://github.com/Automattic/wp-calypso/blob/5a0701af20f215508dd9e606f232ee5edb59270d/apps/full-site-editing/full-site-editing-plugin/blog-posts-block/editor.js#L41).

iamtakashi commented:
Thank you all for working on this! We'd love to be able to use more than one instance of the block soon, and I was wondering how this PR is going.

noahtallen commented:
> Is that "Send hook to Mission Control" failure something I need to worry about?

(not sure if I responded to this yet!)

@obenland No, it was a bug which could happen the first time a PR was created. I did fix it about a week ago. :) But since the PR already exists now, any future pushes or rebases would not trigger the same error.

noahtallen commented:
Also, if we need to update the block, this PR is ready for testing: https://github.com/Automattic/wp-calypso/pull/40144

noahtallen commented:
BTW, once this PR is rebased on `master`, the block version should be 1.1, which might help things :)

obenland commented:
@WunderBart https://github.com/Automattic/newspack-blocks/pull/421 addresses your concerns about the dupes in the editor. I'll merge this PR since it's unrelated from a newspack block update. That way we'll be ready for the update once it makes it to FSE.
